crates/nu-std/CONTRIBUTING.md

```diff
@@ -0,0 +1,191 @@
+# Contributing
+
+Welcome to the Nushell standard library and thank you for considering contributing!
+
+## Ideas for the standard library
+
+If you've got a great idea, or just want to contribute to open source
+by working on the Nushell standard library, 
+we invite you to talk to the team before you start coding. 
+You'll find we're friendly, passionate about Nushell and always open to new ideas!  
+
+You'll generally find the team members on
+[Discord standard-library channel](https://discord.com/channels/601130461678272522/1075541668922658868), 
+and can have preliminary discussions there to clarify the issues involved.
+
+You can open a [Github issue](https://github.com/nushell/nushell/issues/new/choose) 
+to have a more focused discussion of your idea.  
+
+Generally, we think the standard library should contain items that are 
+relevant to most/all Nushell users regardless of the application space they're working in. 
+If your idea isn't quite so broadly applicable, 
+consider publishing it in [nu_scripts](https://github.com/nushell/nu_scripts).
+
+Preliminary discussions should focus on the *user benefit* your idea would provide.  
+How many users will be affected by your idea, how would (a good implementation of) it 
+improve their Nushell experience?
+Given consensus on the user benefit, the team will be motivated to 
+help you create, deploy and maintain a solution long term.
+
+## Lifecycle of a change
+
+1. Verify the team thinks your idea is potentially relevant and useful, as above.
+1. If it's more than a simple bug fix, open a placeholder PR 
+as soon as you get started and mark it as work in progress (WIP).
+This will alert other contributors that you're working in this area and 
+advertise roughly what scope of changes you're thinking of.
+See [below](the_pr) for details.
+1. Get things working in your local development environment.  
+If you have questions along the way, you can post a question in your PR 
+or have a more casual discussion with Nushell fans on 
+[Discord implementation-chat channel](https://discord.com/channels/601130461678272522/615962413203718156)
+1. When you get to an appropriate state of doneness, push your changes to the PR and remove the WIP flag.
+2. Team members and other contributors will then review your PR.  
+Respond to any review comments they raise and resolve them one way or another.
```

Comment:
i you mean "solve the review threads when you, as the author of the PR, thinks you have addressed them", i'd strongly discourage that :yum: 

as a reviewer, i find this super confusing and tideous when someone else than me, either the author or another reviewer, closes a thread i've opened for me :confused: 
i've opened the thread, so i'd like to know if the changes have been properly addressed :smirk: 
when a thread is closed by someone else, it does not appear anymore in the review feed and the reviewer would have to scroll all the resolved threads to find what has been answered, if the changes have been approved, etc, etc...

of course, that is my experience and i find that very confusing myself, but i'd really like not mentionning that opportunity :eyes: 

=> i also think specifying in the thread what commit(s) should addressed the issue is very sweet for reviewers :relieved:

Reply:
My use of "resolve" is confusing, maybe "address" is closer to the mark?  I mean that the owner of the PR should change the code or convince the reviewer it's fine as is.
Also, is it the norm on this project that the reviewer who opens the thread can be relied on to close it when satisfied?  Github allows other participants to "resolve" the thread, and I've seen projects where the owner of the PR is the only one who's motivated to clean up the threads.

Reply:
> My use of "resolve" is confusing, maybe "address" is closer to the mark? I mean that the owner of the PR should change the code or convince the reviewer it's fine as is. 
>
oh yeah perfect!
then i'd keep "address" for this very action :wink: 

as "resolve" is the action *GitHub* refers to on the button to "close" the threads, that might be misleading?

> Also, is it the norm on this project that the reviewer who opens the thread can be relied on to close it when satisfied?
>
i do not know if there's a norm here :thinking: 
maybe in the main `CONTRIBUTING.md`?

these are my two cents :relieved: 
i really find that super confusing not seing my threads where i left them :confused: 

> **Note**
> `nushell` does not require approvals nor thread resolution to merge a PR, so we do not even really *need* to close threads before merging.
> i see them as indications for another maintainer that wants to answer the question "can i merge this PR" => an opened thread without clear closure might indicate a non-quite-ready PR

> Github allows other participants to "resolve" the thread, and I've seen projects where the owner of the PR is the only one who's motivated to clean up the threads.
>
imo, that looks like a pretty bad practice :thinking: 

i think review threads ARE the review itself.
they are the place where maintainers and reviewers discuss with the author what needs to be changed, what could be changed.
if the author closes the threads, that looks almost like a hack where they say "i wont' work on this and you won't see this unless you scroll all your closed threads" => that could make the "thread resolution" requirement branch protection rule happy even though nothing has been addressed :open_mouth:

it's like doing a rebase / force-push without a reason => that disqualifies all the previous review and commits because i do not have any guarantee as a reviewer that the rebase did not break any thing :yum: 

in the end i'd say three things
- there is no consensus on this, this is from my personal review experience
- i find closing my threads for me really confusing and a waste of time as i need to go and get them back...
- i think the responsibility to close a thread is on the person who opened it (or maybe a very high-permission maintainer who really knows what their doing :/)

Reply:
(not targeting your force-push here @bobhy, this was a good one :wink:)

Reply:
i'm really just expressing very general review guidelines but i can not resist :eyes:
i think a good review workflow is really important :yum: 

TL;DR: changing "resolve" to "address" in this document would be just perfect :blush: